### PR TITLE
56139 convert datetime from UTC to local on appointment responses

### PR DESCRIPTION
**Pull request by cferris32**
## Summary

- This PR is to add functionality to vets-api to convert appointment times from UTC to local on the responses for index/show. This is mainly handled by the new method `convert_utc_to_local_time` which does exactly that. We also supplemented the calls to retrieve facility data with additional error handling to make it clear when the timezone could not be adjusted due to lack of facility data returned.

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#56139

## Testing done

- tested new `convert_utc_to_local_time` method
- tested error handling for facility calls when unable to fetch facility/timezone data
- new unit tests for appointments_service
- regression tested integration/request specs

## What areas of the site does it impact?
This impacts the format of the appointment time returned from vets-api

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by ajmagdub — APPROVED** [review]
LGTM

**Review by simiadebowale — APPROVED** [review]

**Review by simiadebowale — DISMISSED** [review]

**Review by ryan-mcneil — DISMISSED** [review]
LGTM

**Review by ericboehs — DISMISSED** [review]

**Review by rmtolmach — DISMISSED** [review]

**Comment by jperk51** [comment]
Any chance of moving this logic into the service so it gets picked up by anyone using the existing get_appointments logic? Would be nice to not have to copy your logic for mobile and would ensure that anyone integrating with it in the future would automatically get this functionality.

**Comment by cferris32** [comment]
> Any chance of moving this logic into the service so it gets picked up by anyone using the existing get_appointments logic? Would be nice to not have to copy your logic for mobile and would ensure that anyone integrating with it in the future would automatically get this functionality.

Will refactor then update here once that change is ready

**Comment by jperk51** [comment]
@cferris32 once this is approved please hold off on merging until after the daily prod deploy so that mobile can have at least 24 hours to test it before the next prod deploy in case it needs to be reverted.

**Comment by cferris32** [comment]
> @cferris32 once this is approved please hold off on merging until after the daily prod deploy so that mobile can have at least 24 hours to test it before the next prod deploy in case it needs to be reverted.

Will do, all set for review

**Comment by Tonksthebear** [comment]
Also just a note for any other mobile members looking at this, some of the things I changed in our directory in this is temporary, such as the `time_to_datetime`. After this pr gets merged, I'll be removing any code where we do datetime conversions

**Review by ajmagdub — COMMENTED** [review]
LGTM, you might want to remove the actual JWT with 'stubbed_token' in the VCR cassettes

File: modules/vaos/app/services/vaos/v2/appointments_service.rb
Comment (kpethtel) [inline modules/vaos/app/services/vaos/v2/appointments_service.rb:124]: Given that this is happening twice, it seems like it could've happened in the service.

**Review by kpethtel — DISMISSED** [review]
LGTM 👍

File: modules/vaos/app/controllers/vaos/v2/appointments_controller.rb
Comment (Tonksthebear) [reply modules/vaos/app/controllers/vaos/v2/appointments_controller.rb:267]: I didn't add this code if you were referencing me 🙃

**Review by simiadebowale — APPROVED** [review]

File: modules/vaos/app/controllers/vaos/v2/appointments_controller.rb
Comment (cferris32) [reply modules/vaos/app/controllers/vaos/v2/appointments_controller.rb:267]: We wanted to provide more specific messaging for when facility info couldn't be retrieved, also meaning the timezone wouldn't be converted. We saw it as providing slightly more value than just checking for nil in this situation, in case the location field is nil for other reasons. The website has plans to surface this message accordingly, but mobile can certainly use it differently if y'all would prefer. Either way, I'll shuffle that constant up in the file 🙂

**Review by Tonksthebear — APPROVED** [review]
LGTM

**Review by ajmagdub — APPROVED** [review]
LGTM

**Pull request merged**
